src/main/java/net/snowflake/client/core/arrow/fullvectorconverters/ListVectorConverter.java

```diff
@@ -58,8 +65,7 @@ protected FieldVector convertVector()
       FieldVector convertedDataVector =
           ArrowFullVectorConverterUtil.convert(
               allocator, dataVector, context, session, timeZoneToUse, 0, valueTargetType);
-      // TODO: change to convertedDataVector and make all necessary changes to make it work
-      ListVector convertedListVector = initVector(vector.getName(), dataVector.getField());
+      ListVector convertedListVector = initVector(vector.getName(), convertedDataVector.getField());
```

Comment:
This is the line that required the whole change - if we don't keep nullability properly, mapVector will complain about it's grandchildren being nullable. As the grandchildren may be of any type, all converters were affected.

Reply:
let's add some tests to show the improvement

Reply:
Added a unit test checking if FieldTypes behave as expected when converting a MapVector